web/src/main/java/org/akaza/openclinica/control/admin/UpdateSubjectServlet.java

```diff
@@ -276,56 +276,56 @@ else if ( currentStudy.getStudyParameterConfig().getCollectDob().equals("2")){
 
         // uniqueIdentifier must be unique in the system
         if (currentStudy.getStudyParameterConfig().getSubjectPersonIdRequired().equals("required")
-    			|| currentStudy.getStudyParameterConfig().getSubjectPersonIdRequired().equals("optional")){
+                || currentStudy.getStudyParameterConfig().getSubjectPersonIdRequired().equals("optional")){
 
-	        String uniqueIdentifier = fp.getString("uniqueIdentifier");
-	        if (currentStudy.getStudyParameterConfig().getSubjectPersonIdRequired().equals("required") &&
-	        		!(subject.getUniqueIdentifier() == null || subject.getUniqueIdentifier().isEmpty() ) &&
-	        		(uniqueIdentifier == null || uniqueIdentifier.isEmpty())) {
-	        	Validator.addError(errors, "uniqueIdentifier", resexception.getString("field_not_blank"));
+            String uniqueIdentifier = fp.getString("uniqueIdentifier");
+            if (currentStudy.getStudyParameterConfig().getSubjectPersonIdRequired().equals("required") &&
+                    !(subject.getUniqueIdentifier() == null || subject.getUniqueIdentifier().isEmpty() ) &&
+                    (uniqueIdentifier == null || uniqueIdentifier.isEmpty())) {
+                Validator.addError(errors, "uniqueIdentifier", resexception.getString("field_not_blank"));
+                
+            }
+             if (uniqueIdentifier != null && !uniqueIdentifier.isEmpty()) {
                 
+                 
+                if ( uniqueIdentifier.length() > 255){
```

Comment:
Can we define a constant for this (255) some where so we can change it we need to?